src/retrieval/reranking_algorithms.py

```
@@ -0,0 +1,243 @@
+            
+            # Combine all factors
+            enhanced_score = (
+                result.similarity_score * 0.4 +  # Base similarity
+                overlap_score * 0.2 +           # Word overlap
+                length_score * 0.2 +            # Content length
+                position_score * 0.1 +          # Position preference
+                interaction_score * 0.1         # Query-document interaction
+            )
+            
+            result.relevance_score = min(1.0, enhanced_score)
+            scored_results.append(result)
```

Ruff (0.13.1)

Undefined name `scored_results`

(F821)